### PR TITLE
feat(stream): support stream subsystem discovery

### PR DESCRIPTION
**Pull request by adugeek**
Signed-off-by: adugeek <root@libssl.com>

### What this PR does / why we need it:

resolve: #4469 

 int function init.lua:stream_init_worker() not call discovery.init_worker(), 
 so, we can't use discovery in stream subsystem.
 by this pr, we can slove it
 of course, the current and future discovery components need to ensure they can work in the stream subsystem 

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by tokers** [comment]
@adugeek Please add some test cases to verify the changes are effective.

File: apisix/init.lua [hunk 1]
```diff
@@ -823,6 +823,11 @@ function _M.stream_init_worker()
         core.config.init_worker()
     end
 
+    local discovery = require("apisix.discovery.init").discovery
+    if discovery and discovery.init_worker then
+        discovery.init_worker()
+    end
```
Comment (spacewander) [inline apisix/init.lua:829]: You can add a test with the DNS discovery: https://github.com/apache/apisix/blob/07bfe5b9c03ad361cb71541eb02fd661f1323213/t/discovery/dns/sanity.t#L91

You can setup it with https://github.com/apache/apisix/blob/07bfe5b9c03ad361cb71541eb02fd661f1323213/utils/set-dns.sh#L33

File: t/discovery/dns/sanity.t [hunk 2]
```diff
@@ -284,3 +284,34 @@ qr/proxy request to \S+/
...
+      discovery_type: dns
+      type: roundrobin
+      id: 1
+--- stream_enable
+--- stream_request eval
+"GET /hello HTTP/1.1"
+--- stream_response
+hello world
+--- no_error_log
+[error]
```
Comment (tokers) [inline t/discovery/dns/sanity.t:317]: Should add the EOS symbol. Please check your editor.

**Comment by tokers** [comment]
@adugeek Please check out the CI status.

**Comment by adugeek** [comment]
@tokers sorry, busying in the past week.
 and it takes a little time to learn "how to write test case"

File: t/discovery/dns/sanity.t [hunk 3]
```diff
@@ -284,3 +284,34 @@ qr/proxy request to \S+/
...
+--- yaml_config
+apisix:
+    node_listen: 1984
+    config_center: yaml
+    enable_admin: false
+    enable_resolv_search_option: false
+    stream_proxy:
+        tcp:
+            - addr: 1985
+discovery:                        # service discovery center
```
Comment (Yiyiyimu) [inline t/discovery/dns/sanity.t:298]:
```suggestion
discovery:
```

**Comment by Yiyiyimu** [comment]
@adugeek CI failed, please recheck

**Pull request closed without merging**